src/DataTypes/TimezoneMixin.h

```diff
@@ -29,4 +37,17 @@ class TimezoneMixin
 
     const DateLUTImpl & time_zone;
     const DateLUTImpl & utc_time_zone;
+
+private:
+    static const String & checkTimezoneName(const String & timezone_name)
+    {
+        const char * forbidden_patterns[] = {"/", "../", "./", "~/"};
+        for (const auto & pattern : forbidden_patterns)
+        {
+            if (timezone_name.starts_with(pattern))
```

Comment:
Can it contain a forbidden pattern in the middle?

Reply:
Good catch, we should try to normalize the path before this check. I will do it

Reply:
I didn't find a simple way to normalize a path. I think checking `"../"` substring will be enough.